### DISCUSSION TITLE
Gemini and Gemtext integration

### DISCUSSION THREAD

**Discussion by xplosionmind** [11ty/eleventy#2309] [Enhancement Queue]
It would be awesome if it could be created a plugin that generates a website accessible from the [Gemini protocol](https://gemini.circumlunar.space), by generating `.gmi` files in [Gemtext](https://gemini.circumlunar.space/docs/gemtext.gmi), that the user can manage to deploy in parallel with the “standard” website. [md2gemini](https://github.com/makeworld-the-better-one/md2gemini) might be an interesting starting point.

**Comment by pdehaan** [comment]
I'm not too sure how you could build that. I imagine it would have a lot of edge cases around using LiquidJS tags or shortcodes/filters in your .md files.

One approach I was thinking of is you could try creating a [transform](https://www.11ty.dev/docs/config/#transforms) or something that would convert from .html to gemini, but you might need to do some complex DOM parsing since the transforms run really late in the pipeline when all the content has been processed (so the layouts would have all been applied). So you might need to then just extract all rendered HTML content in some arbitrary `<main data-gemini="true">...</main>` tag and see if you can find a Node.js based HTML to Gemini converter that would extract and process the data and then manually write the files to a modified `this.outputPath` and still return the original content from the transformer so both the .gmi and .html files would be written.

Another possible approach that came to mind was having all your content in .md files, then creating some special collection which converts the .md content to .gmi files and then use pagination to loop over the files and make sure they get written without a layout applied.
Might work, but you'd need to search npm and see what existing tools exist and possibly have to write them yourself since this didn't seem to be a very common thing based on my 30 second npm searching. Worst case you could probably call Python scripts from your Node using the built-in `child_process` module.
